esp-hal-common/src/usb_serial_jtag.rs

```diff
@@ -51,23 +52,25 @@ impl<'d> UsbSerialJtag<'d> {
     }
 
     /// Write data to the serial output in chunks of up to 64 bytes
-    pub fn write_bytes(&mut self, data: &[u8]) -> Result<(), Error> {
+    pub fn write_bytes(&mut self, data: &[u8]) -> Result<usize, Error> {
         let reg_block = self.usb_serial.register_block();
 
+        let mut count = 0;
         for chunk in data.chunks(64) {
-            unsafe {
-                for &b in chunk {
-                    reg_block.ep1.write(|w| w.rdwr_byte().bits(b.into()))
-                }
-                reg_block.ep1_conf.write(|w| w.wr_done().set_bit());
+            for byte in chunk {
+                reg_block
+                    .ep1
+                    .write(|w| unsafe { w.rdwr_byte().bits(*byte) });
+                count += 1;
+            }
+            reg_block.ep1_conf.write(|w| w.wr_done().set_bit());
 
-                while reg_block.ep1_conf.read().bits() & 0b011 == 0b000 {
-                    // wait
-                }
+            while reg_block.ep1_conf.read().bits() & 0b011 == 0b000 {
+                // wait
             }
         }
 
-        Ok(())
+        Ok(count)
```

Comment:
This function already guarantees that it will send all the data, so I think we can just return `Ok(data.len())` here.